sdk/azidentity/azidentity.go

```diff
@@ -148,22 +152,23 @@ type pipelineOptions struct {
 }
 
 // setAuthorityHost initializes the authority host for credentials.
-func setAuthorityHost(authorityHost string) (string, error) {
-	if authorityHost == "" {
-		authorityHost = AzurePublicCloud
-		// NOTE: we only allow overriding the authority host if no host was specified
+func setAuthorityHost(authorityHost AzureAuthorityHost) (string, error) {
+	host := string(authorityHost)
+	if host == "" {
 		if envAuthorityHost := os.Getenv("AZURE_AUTHORITY_HOST"); envAuthorityHost != "" {
-			authorityHost = envAuthorityHost
+			host = envAuthorityHost
+		} else {
+			host = string(AzurePublicCloud)
```

Comment:
You could move this before the call to `os.Getenv()` to avoid the `else` clause.

Reply:
That's how it was, I changed it because the unnecessary second assignment when the `if` evaluates true seemed strange to me. Is avoiding `else` like this typical Go style?